### PR TITLE
fix(numpy): add context/return_scalar params to Tensor.__array_wrap__

### LINKED ISSUE #180657: BUG: Avoid NumPy 2 deprecation warning in `__array_wrap__`
### 🐛 Describe the bug

In NumPy 2, NumPy added a `DeprecationWarning` on `__array_wrap__` to extend the information passed in and avoid an indefinite `try/except` chain.

This means that the e.g. the following gives a warning:
```
import warnings

import numpy as np
import torch

warnings.filterwarnings("error")  # with this an error

np.delete(torch.arange(10), 0)
```

Now, the solution is straight forward: Just add the two new arguments and ignore them.
(torch has no scalars, and the context is info that was always there and you don't need).  That is `__array_wrap__(obj, context=None, return_scalar=None)`.

I'll guess end-users use this path but because of how `DeprecationWarning`s work they won't actually see them normally, so it makes sense it wasn't spotted.
It would be nice to escalate the warning to something visible, but I would prefer if by that time torch has the fix.

### Versions

current main.

cc @mruberry @rgommers

### PR DESCRIPTION
**Pull request by ogulcanaydogan**
Fixes #180657

NumPy 2 extended the `__array_wrap__` protocol with two new keyword arguments (`context` and `return_scalar`) and emits a `DeprecationWarning` when the old two-argument form is called. This surfaces as a warning (or error with `warnings.filterwarnings("error")`) in common operations such as `np.delete(tensor, ...)`.

**Fix:** Accept the two new parameters and ignore them. PyTorch has no scalars, and the `context` info is not needed downstream.

Changes:
- `torch/_tensor.py`: `def __array_wrap__(self, array)` → `def __array_wrap__(self, array, context=None, return_scalar=None)`
- `torch/overrides.py`: update the testing-override lambda to match the new signature
- `test/test_numpy_interop.py`: regression test — verifies `np.delete(tensor, ...)` does not raise `DeprecationWarning`

**Testing:** The new test (`test_array_wrap_numpy2_no_deprecation_warning`) runs on CPU and calls `np.delete` with `warnings.simplefilter("error", DeprecationWarning)` so any future regression would be caught immediately.

### CODE REVIEW

**Comment by ogulcanaydogan** [comment]
Friendly bump — CI is green and labels look good. Happy to address any feedback.

**Comment by ogulcanaydogan** [comment]
Gentle ping on this one. It's a small test-only refactor (parallel subtests) and CI is green. Happy to rebase if it has drifted.

**Comment by ogulcanaydogan** [comment]
Looks like this got fixed in #187190, so this is no longer needed. Closing.
